### PR TITLE
Generate and publish SDK API reference from Travis builds of master branch

### PR DESCRIPTION
**Pull request by ycombinator**
This PR generates the API reference for the PHP SDK and publishes it to the `gh-pages` branch, under the `docs/api` folder. Users would be able to view this API reference at http://php-opencloud.com/docs/api.

This PR deliberately does not create any links to that location from other HTML files in the `gh-pages` branch. This is because I suspect we will want to review the generate API reference and make additions before making it discoverable by end-users.

Please note that the reference generation and publish process will not actually occur until the changes in this PR are merged into the `master` branch. This is to ensure that the reference is always generated only from the PHP source code in the `master` branch.

Attn: @jamiehannaford, @gecampbell

### CODE REVIEW

File: docs/generate.sh [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
+# to the master branch, do nothing.
+if [ "$TRAVIS_BRANCH" != "master" ]; then
+  exit 0
 fi
 
-if [ ! -d docs ]; then
-    echo "No docs/ directory found; run this script from the top directory"
-    exit;
+# Generate the API references
+rm -rf $API_DOCS_DIR && \
```
Comment (jamiehannaford) [inline docs/generate.sh]: Will `docs/api` ever exist in master?

File: .travis.yml [hunk 2]
```diff
@@ -30,3 +30,6 @@ notifications:
     - jamie.hannaford@rackspace.com
     - glen.campbell@rackspace.com
     - shaunak.kashyap@rackspace.com
+
+env:
+  - secure: "bdrUeYb3nSGgBB+QtDZxUHVPw6B/wjb3KXLm8TgonWrQm4GPeWKK29qhmDnFZmQjwQPfuebe7wAk1ZxGoZKbEiELVpJJ+8XYVOt6W/6V53H31JL6FqiIE5+7qBwDe+9ziveM6GcTXHT1GI5mUeACIbeBDPZaNubIJH3U6MPim64="
```
Comment (jamiehannaford) [inline .travis.yml]: :+1: didn't know you could do this

**Comment by jamiehannaford** [comment]
LGTM apart from that 1 line comment

File: docs/generate.sh (hunk 1 above)
Comment (ycombinator) [inline docs/generate.sh]: No, it shouldn't. I think this line can go. Removing.

**Comment by jamiehannaford** [comment]
:ship:

**Pull request merged**